### PR TITLE
feat: global cache manager

### PR DESCRIPTION
**Pull request by iammuho**
Related issue: https://github.com/mergeability/mergeable/issues/502

--- 

To solve this issue and create functionality for the future, I've created a cache manager to be able to make the logic out of other controllers. With this feature, if anyone needs a cache functionality, they don't have to care about the store but just use the adapter functions. Cache manager will handle the rest depends on the ENV variables(or global settings future) for the CACHE_STORAGE var.

With this feature, we are introducing new ENVIRONMENT variables and we cannot place them to settings, because cacheManager works before settings file parsed. New environment variables are placed onto `.env.example` file

Feature list:

- [x] IN-Memory cache is working
- [x] Redis cache is working

### CODE REVIEW

File: lib/configuration/configuration.js
Comment (shine2lay) [inline lib/configuration/configuration.js:6]: can we please use either `const` or `let` instead of `var` for consistency sake?

**Review by shine2lay — APPROVED** [review]
LGTM, have one tiny suggested change.
Just food for thought, should we add this to `settings` file as well once it's been merged?

File: lib/cache/cache.js
Comment (shine2lay) [inline lib/cache/cache.js:10]: should `refreshThreshold` be a configurable variable?

File: lib/cache/cache.js
Comment (shine2lay) [inline lib/cache/cache.js:13]: what happens if there are more than 100 repos, what happens then? is it only storing the last 100 repo settings?

Comment (iammuho) [reply lib/cache/cache.js:13]: hey @shine2lay,

Check the stackoverflow answer: https://stackoverflow.com/a/24230894

File: lib/cache/cache.js
Comment (iammuho) [reply lib/cache/cache.js:10]: Hey you are right, I've moved every variable to environment variables. Defaults has been defined by the parent package already,

```
max: 500
ttl: null
```

**Comment by iammuho** [comment]
> LGTM, have one tiny suggested change.
> Just food for thought, should we add this to `settings` file as well once it's been merged?

After you merge the settings pages, I'll modify here to use `global settings` for cache configuration.

**Comment by iammuho** [comment]
Hey @shine2lay I've migrated settings to the `cacheManager` too and waiting for your reviews :)

**Review by jusx — APPROVED** [review]
Looks good. I like that your choice of switching to `cache-manager`

**Comment by iammuho** [comment]
hey @jusx or @shine2lay , waiting for your merge. Fixed the conflicts

**Comment by shine2lay** [comment]
LGTM, merging

**Pull request merged**
